Pass cache variables to ext_mpifx

pecchia opened the pull request:
I still had some problem with compilation because I usually have to set my custom.cmake in order to convince cmake to find the correct compiler. This fix allows to pass some cache variables down to mpifx compilation. 

General question. Does this impact dftb+ compilation. I did not try yet

Alex

Commit:
Pass cache variables to ext_mpifx

gpenazzi commented:
> General question. Does this impact dftb+ compilation. I did not try yet

Good question :) I remembered that dftb+ was having a custom cmake bypassing ours so I didn't check. But it's not the case, so either it's never been so or it has changed at some point. I am trying to check

gpenazzi committed:
Make the cmake more project-friendly:
- Append to modules instead of overwriting
- Respect user choice ot MPIFX_LIB
- Rename LIBRARY to LIB

gpenazzi commented:
I checked, DFTB+ needs some more fixes anyway because we changed some things in the modules (e.g. unit->units). But anyway I identified a couple of things which would not play well and added a commit here. For me we can merge, anyway we might have to come back on the cmake.
